src/task/NpmScriptTask.ts

```diff
@@ -194,12 +194,14 @@ export class NpmScriptTask {
       context.measures.failedTasks.push({ pkg: info.name, task });
       this.onFail();
 
-      if (!config.continue) {
-        controller.abort();
-        return false;
-      } else {
+      if (config.continue) {
         return true;
       }
+
+      if (!config.safeExit) {
+        controller.abort();
+      }
+      return false;
```

Comment:
Returning `false` here is sufficient to stop other queued tasks from not running?

Reply:
This method is called in the [`runHandler`](https://github.com/microsoft/lage/blob/91749da316f6d2ede5b1981b92f819a7ff5eea2a/src/task/taskRunner.ts#L136) which is passed to the task-scheduler [pipeline](https://github.com/microsoft/lage/blob/91749da316f6d2ede5b1981b92f819a7ff5eea2a/src/task/taskRunner.ts#L106) as a task.

I think that the contract with the task-scheduler is that if a task `run` function returns `false`, it will skip the rest of the queued tasks.

Reply:
Yes, you're right. If the task passed to task-scheduler returns false, task-scheduler will throw and catch and set bail = true, and any subsequent tasks will skip if bail is set to true.

https://github.com/microsoft/task-scheduler/blob/3c4f86efeb249a902ac0d563b8e3c9526d791f12/src/pipeline.ts#L36